### PR TITLE
glogger: Allow postponed initialization of Sender

### PR DESCRIPTION
**Pull request by mirikl**

### CODE REVIEW

**Review by netaneld122 — DISMISSED** [review]
LGTM

**Review by DanielShaulov — DISMISSED** [review]

**Review by d3dave — COMMENTED** [review]
Please update the docstring accordingly.

**Review by netaneld122 — APPROVED** [review]
LGTM, I want @DanielShaulov to have a look as well

File: glogger/handler.py [hunk 1]
```diff
@@ -51,21 +51,31 @@ def __init__(
...
 
         self.stdout_logger = get_stdout_logger()
         self.jsonify = JSONEncoder(separators=(",", ":"), default=repr).encode  # compact, no whitespace
         self.messages_buffer = MessagesBuffer(max_total_length, overflow_drop_factor)
         self.messages_buffer.head_serial_no = continue_from
 
+        self.sender: Optional[Sender] = None
+        self.init_sender(sender)
+
+    def init_sender(self, sender: Optional[Sender]) -> None:
```
Comment (netaneld122) [inline glogger/handler.py:64]: Actually, let's accept a non-optional `Sender` here and check if it's None in the __init__ function instead

**Comment by DanielShaulov** [comment]
What happened here? What was wrong with approved changes for Yesterday?

**Comment by mirikl** [comment]
> What happened here? What was wrong with approved changes for Yesterday?

@DanielShaulov , after talking with @netaneld122 it seems that this change is cleaner then the other one.
The change in sender is more complicated because we need to perform the change we did for `auth_token` to `server_address` also.

**Review by DanielShaulov — APPROVED** [review]

**Pull request merged**